### PR TITLE
perf(GraphQL): RHICOMPL-2426 reduce the queries for systems resources

### PR DESCRIPTION
**Pull request by skateman**
* Renamed `GET_SYSTEMS_ONLY` to `GET_SYSTEMS_WITH_POLICIES` 
* Replaced `GET_SYSTEMS` in profile details page with `GET_SYSTEMS_WITH_POLICIES`
* Renamed `GET_SYSTEMS` to `GET_SYSTEMS_WITH_REPORTS`
* Replaced `GET_SYSTEMS_WITHOUT_FAILED_RULES` with `GET_SYSTEMS_WITH_POLICIES`
* Deleted the obsolete `GET_SYSTEMS_WITHOUT_FAILED_RULES`

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

**Comment by skateman** [comment]
@bastilian can you please take a look?

File: src/SmartComponents/PolicyDetails/PolicySystemsTab.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ const PolicySystemsTab = ({ policy }) => (
       Columns.SsgVersion,
     ]}
     showOsMinorVersionFilter={[policy.osMajorVersion]}
-    query={GET_SYSTEMS}
+    query={GET_SYSTEMS_WITH_PROFILES}
```
Comment (bastilian) [inline src/SmartComponents/PolicyDetails/PolicySystemsTab.js:20]: This should be `GET_SYSTEMS_WITH_PROFILES` as it requires the `testResultProfiles` to get the SSG version.

local vs. stage:
Screenshot 2022-05-03 at 12 18 12
:

**Review by bastilian — CHANGES_REQUESTED** [review]
Thank you! Awesome @skateman! LGTM, other than the one query on the Policy details.

File: src/SmartComponents/PolicyDetails/PolicySystemsTab.js (hunk 1 above)
Comment (skateman) [reply src/SmartComponents/PolicyDetails/PolicySystemsTab.js:20]: Okay, the naming is wrong and it should be `GET_SYSTEMS_WITH_POLICIES` and I guess here it should be `GET_SYSTEMS_WITH_REPORTS`.

Comment (bastilian) [reply src/SmartComponents/PolicyDetails/PolicySystemsTab.js]: 🤦 oh, yeah. my tired eyes don't go well with copy&pasting. :D

**Review by bastilian — APPROVED** [review]
Thank you @skateman! All looks good now!

**Pull request merged**